ambry-messageformat/src/main/java/com.github.ambry.messageformat/MessageFormatRecord.java

```diff
@@ -1469,6 +1471,97 @@ public static CompositeBlobInfo deserializeMetadataContentRecord(DataInputStream
       return new CompositeBlobInfo(chunkSize, totalSize, keys);
     }
   }
+
+  /**
+   *  - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+   * |         |              |            |            |          |            |            |          |
+   * | version |  total size  | # of keys  |  size of   |   key1   |  size of   |    key2    |  ......  |
+   * |(2 bytes)|  (8 bytes)   | (4 bytes)  | key1 blob  |          | key2 blob  |            |  ......  |
```

Comment:
Is the totalSize field meant for sanity checking that `sum(chunksizes) == totalsize` or are there other places where it is used?

Reply:
Yeah it's only used for checking